bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/items/ItemRegistryOSGiTest.java

```diff
@@ -86,6 +87,7 @@ public void setUp() {
 
     @After
     public void tearDown() {
+        unregisterService(eventSubscriber);
```

Comment:
Tests which do not use the `eventSubscriber` will run into an NPE here. You should only unregister it if it was set before...

https://github.com/eclipse/smarthome/blob/master/bundles/test/org.eclipse.smarthome.test/src/main/java/org/eclipse/smarthome/test/java/JavaOSGiTest.java#L285

Reply:
On my machine maven produces this output:

```
Results :

Tests run: 21, Failures: 0, Errors: 0, Skipped: 0

[INFO] All tests passed!
```

Reply:
The instance `eventSubscriber` is a mock which is none-null in every test. It just gets registered in a few of them.

Reply:
Strange, because this test:

https://github.com/eclipse/smarthome/blob/a6140cbd8db5d50e22dc33975d36705e453bf411/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/items/ItemRegistryOSGiTest.java#L92-L97

does not set the `eventSubscriber` and according to

https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/items/ItemRegistryOSGiTest.java#L64

it is `null`.

Unless the `initMocks(this)` call sets it to something maybe?

https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/items/ItemRegistryOSGiTest.java#L68

That would explain it.

Reply:
Damn you were quicker... Thanks.